pkg/controller/k8s/pod_controller.go

```diff
@@ -70,28 +70,50 @@ func (r *PodReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{}, nil
 	}
 
-	// Add new pod
 	// skip host network
 	if pod.Spec.HostNetwork {
 		return ctrl.Result{}, nil
 	}
 
 	endpoint := v1alpha1.Endpoint{}
-	endpoint.Name = endpointName
-	endpoint.Namespace = req.Namespace
-	endpoint.Spec.VID = 0
-	endpoint.Spec.Reference.ExternalIDName = "pod-uuid"
-	endpoint.Spec.Reference.ExternalIDValue = utils.EncodeNamespacedName(types.NamespacedName{
-		Name:      endpointName,
+	endpointReq := types.NamespacedName{
 		Namespace: req.Namespace,
-	})
-
-	// submit creation
-	if err := r.Create(ctx, &endpoint); err != nil {
-		if errors.IsAlreadyExists(err) {
-			return ctrl.Result{}, nil
+		Name:      endpointName,
+	}
+	err := r.Get(ctx, endpointReq, &endpoint)
+	if errors.IsNotFound(err) {
+		// add new pod
+		endpoint.Name = endpointName
+		endpoint.Namespace = req.Namespace
+		endpoint.Spec.VID = 0
+		endpoint.Spec.Reference.ExternalIDName = "pod-uuid"
+		endpoint.Spec.Reference.ExternalIDValue = utils.EncodeNamespacedName(types.NamespacedName{
+			Name:      endpointName,
+			Namespace: req.Namespace,
+		})
+		endpoint.ObjectMeta.Labels = map[string]string{}
+		for key, value := range pod.ObjectMeta.Labels {
+			endpoint.ObjectMeta.Labels[key] = value
+		}
+		// submit creation
+		if err := r.Create(ctx, &endpoint); err != nil {
+			klog.Errorf("create endpoint %s err: %s", endpointName, err)
+			return ctrl.Result{}, err
+		}
+	}
+	if err == nil {
+		// update pod
+		for key, value := range pod.ObjectMeta.Labels {
+			endpoint.ObjectMeta.Labels[key] = value
 		}
-		klog.Errorf("create endpoint %s err: %s", endpointName, err)
+		// submit update
+		if err := r.Update(ctx, &endpoint); err != nil {
+			klog.Errorf("update endpoint %s err: %s", endpointName, err)
+			return ctrl.Result{}, err
+		}
+	}
+	if err != nil && !errors.IsNotFound(err) {
+		klog.Errorf("Get endpoint error, err: %s", err)
 		return ctrl.Result{}, err
 	}
```

Comment:
if err == nil
else if errors.IsNotFound(err)
else {}
return

Reply:
if else is forbidden in golint, which suggests to use switch. 
So, this part has to split into three if.

Reply:
you can return at the end of if.
```
if err == nil {
    // handle
    return nil
}
if errors.IsNotFound(err) {
    // handle
    return nil
}
return err
```

Reply:
then we can use switch case, return is the same in all cases